### PR TITLE
Add some details of custom CA certs

### LINKED ISSUE #3642: Add some details about custom CA Certs
We support a HTTP boot mode which is the mechanism we use for network install. The user can specify the URL and host etc. Previously this would use a plain HTTP download which is insecure.

We now support adding a custom CA cert enabling HTTPS to be used, which is more secure. Add some details about this.

Please assign to me

### PR DESCRIPTION
**Pull request by peterharperuk**
You can now add your own CA cert to the eeprom to support HTTPS download.

Fixes #3642

### CODE REVIEW

**Comment by peterharperuk** [comment]
@timg236 Can you check this looks ok?

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 1]
```diff
@@ -40,6 +40,55 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+----
+
+On boot it should use HTTPS instead of HTTP.
+
+----
+Loading boot.img ...
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.sig
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.img
+----
+
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:91]:
```suggestion
=== Certificates

For security, Network Install uses HTTPS to download OS images from the Raspberry Pi website. This feature uses our own CA root included in the bootloader to verify the host.

You can add your own custom CA certificate to the EEPROM to securely download images from your own website. Use the `--cacertder` option of the `rpi-eeprom-config` tool to add the DER-encoded certificate.

To ensure that the certificate is not modified, place a hash of the certificate in the EEPROM config settings.

Run the following command to generate a DER-encoded certificate:

[source,console]
----
$ openssl x509 -in your_ca_root_cert.pem -out cert.der -outform DER
----

Then, run the following command to generate a SHA-256 hash of the certificate:

[source,console]
----
$ sha256sum cert.der
----

You should see output similar to the following:

----
701bd97f67b0f5483a9734e6e5cf72f9a123407b346088638f597878563193fc  cert.der
----

Next, update `boot.conf` to include the hash of the certificate:

[source,console]
----
$ sudo rpi-eeprom-config --edit
----

Configure the following settings in the `[gpio8=0]` section, replacing:

* `<your_website>` with your website, e.g. `raspberrypi.com`
* `<path_to_image>` with the path to your OS image hosted on your website, e.g. `path/to/image`
* `<hash>` with the hash value you generated above, e.g. `701bd97f67b0f5483a9734e6e5cf72f9a123407b346088638f597878563193fc`

----
[all]
BOOT_UART=1
POWER_OFF_ON_HALT=0
BOOT_ORDER=0xf461

[gpio8=0]
BOOT_ORDER=0xf7
NET_INSTALL_ENABLED=0
HTTP_HOST=<your_website>
HTTP_PATH=<path_to_image>
HTTP_CACERT_HASH=<hash>
----

NOTE: If you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443.

Finally, use to load everything into EEPROM:

[source,console]
----
$ rpi-eeprom-config -c boot.conf -p public_key.pem -o pieeprom.bin --cacertder cert.der pieeprom.original.bin
$ rpi-eeprom-digest -k private_key.pem -i pieeprom.bin -o pieeprom.sig
----

During network boot, your Raspberry Pi should use HTTPS instead of HTTP. You can check the boot output for the HTTPS URL:

----
Loading boot.img ...
HTTP: GET request for https://yourserver.org:443/path/to/files/boot.sig
HTTP: GET request for https://yourserver.org:443/path/to/files/boot.img
----

```

TL;DR: I made it prettier.

**Review by nathan-contino — APPROVED** [review]
LGTM, with my suggested copy edit!

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc (hunk 1 above)
Comment (lurch) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc]: "Finally, use to load everything..." doesn't sound right?

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 2]
```diff
@@ -40,6 +40,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+NOTE: If you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443.
+
+Finally, use to load everything into EEPROM:
+
+[source,console]
+----
+$ rpi-eeprom-config -c boot.conf -p public_key.pem -o pieeprom.bin --cacertder cert.der pieeprom.original.bin
+$ rpi-eeprom-digest -k private_key.pem -i pieeprom.bin -o pieeprom.sig
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:105]: Are the `public_key.pem` and `private_key.pem` filenames mentioned here supposed to be the same as the `mypubkey.pem` and `myprivkey.pem` mentioned in the "Keys" section of this page?

Also @peterharperuk I just noticed that the "Requirements" section of the page says "To use HTTP boot, update the bootloader to a bootloader dated 10th March 2022 or later." but I guess that's only applicable to the Pi4, and the Pi5 will have a different firmware date in which HTTP boot was made available?

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 3]
```diff
@@ -40,6 +40,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+
+[gpio8=0]
+BOOT_ORDER=0xf7
+NET_INSTALL_ENABLED=0
+HTTP_HOST=<your_website>
+HTTP_PATH=<path_to_image>
+HTTP_CACERT_HASH=<hash>
+----
+
+NOTE: If you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443.
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:98]: Does this mean that if you _don't_ specify a  `HTTP_CACERT_HASH` then Network Install won't download the image using HTTPS, and therefore **also** won't use the custom CA certificate? :thinking:

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 4]
```diff
@@ -40,6 +40,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+Next, update `boot.conf` to include the hash of the certificate:
+
+[source,console]
+----
+$ sudo rpi-eeprom-config --edit
+----
+
+Configure the following settings in the `[gpio8=0]` section, replacing:
+
+* `<your_website>` with your website, e.g. `raspberrypi.com`
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:80]: Perhaps the "e.g. `raspberrypi.com`" should be changed to "e.g. `raspberrypi.com`" so that it matches the `yourserver.org` in the "You can check the boot output for the HTTPS URL" section?
Similarly, perhaps the `path/to/image` in the next line should be updated to match the `path/to/files` in the example boot output?

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 5]
```diff
@@ -40,6 +40,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+$ rpi-eeprom-config -c boot.conf -p public_key.pem -o pieeprom.bin --cacertder cert.der pieeprom.original.bin
+$ rpi-eeprom-digest -k private_key.pem -i pieeprom.bin -o pieeprom.sig
+----
+
+During network boot, your Raspberry Pi should use HTTPS instead of HTTP. You can check the boot output for the HTTPS URL:
+
+----
+Loading boot.img ...
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.sig
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.img
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:113]: This may well be a stupid question (because I'm tired), but how does the firmware know to download both `https://yourserver.org:443/path/to/files/boot.sig` and `https://yourserver.org:443/path/to/files/boot.img` ?
Is `HTTP_PATH` in the eeprom-config set to `path/to/files/boot.img` (with the firmware then doing an extension-swap to get `path/to/files/boot.sig`); or is `HTTP_PATH` set to `path/to/files` (with this then meaning that your hosted OS image **has** to be named `boot.img`) ?

Comment (lurch) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:113]: Ah, https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#HTTP_HOST and https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#HTTP_PATH answer my question (i.e. it's the latter), so perhaps this needs to be made a little bit clearer in this page?

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc (hunk 3 above)
Comment (peterharperuk) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:98]: yes

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc (hunk 2 above)
Comment (peterharperuk) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:105]: yes, this needs a bootloader after 5th April 2024

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc (hunk 4 above)
Comment (peterharperuk) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:80]: I think this gets the message aross?

**Comment by peterharperuk** [comment]
Push a couple of minor edits.

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 6]
```diff
@@ -14,11 +14,13 @@ HTTP_HOST=downloads.raspberrypi.org
...
 ```
 
+boot.img and its associated boot.sig file is a ram disk containg a boot file system, see xref:raspberry-pi.adoc#boot_ramdisk[boot_ramdisk] for more details.
+
 HTTP in the `BOOT_ORDER` will be ignored if secure boot is enabled and xref:raspberry-pi.adoc#HTTP_HOST[HTTP_HOST] is not set.
 
 === Requirements
 
-To use HTTP boot, xref:raspberry-pi.adoc#bootloader_update_stable[update the bootloader] to a bootloader dated 10th March 2022 or later. HTTP boot only works over Ethernet, so you need to connect your Raspberry Pi to your network via an Ethernet cable, e.g. to a socket on the back of your router.
+To use HTTP boot, xref:raspberry-pi.adoc#bootloader_update_stable[update the bootloader] to a bootloader dated 10th March 2022 or later. HTTP boot only works over Ethernet, so you need to connect your Raspberry Pi to your network via an Ethernet cable, e.g. to a socket on the back of your router. Custom CA certificates are supported on 2712 devices with a bootlader dated 5th April 2024 or later.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:23]:
```suggestion
To use HTTP boot, xref:raspberry-pi.adoc#bootloader_update_stable[update] to a bootloader released 10th March 2022 or later. HTTP boot requires a wired Ethernet connection.

To use custom CA certificates, xref:raspberry-pi.adoc#bootloader_update_stable[update the bootloader] to a bootloader released 5th April 2024 or later. Only devices running the BCM2712 CPU support custom CA certificates.
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 7]
```diff
@@ -40,6 +42,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+
+[source,console]
+----
+$ sudo rpi-eeprom-config --edit
+----
+
+Configure the following settings in the `[gpio8=0]` section, replacing:
+
+* `<your_website>` with your website, e.g. `raspberrypi.com`
+* `<path_to_image>` with the path to your OS image hosted on your website, e.g. `path/to/image`
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:83]:
```suggestion
* `<path_to_files>` with the xref:raspberry-pi.adoc#HTTP_PATH[path to your OS image] hosted on your website, e.g. `path/to/files`
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 8]
```diff
@@ -40,6 +42,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+[all]
+BOOT_UART=1
+POWER_OFF_ON_HALT=0
+BOOT_ORDER=0xf461
+
+[gpio8=0]
+BOOT_ORDER=0xf7
+NET_INSTALL_ENABLED=0
+HTTP_HOST=<your_website>
+HTTP_PATH=<path_to_image>
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:96]:
```suggestion
HTTP_PATH=<path_to_files>
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc (hunk 4 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:82]:
```suggestion
* `<your_website>` with xref:raspberry-pi.adoc#HTTP_HOST[your website], e.g. `yourserver.org`
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 9]
```diff
@@ -40,6 +42,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+HTTP_CACERT_HASH=<hash>
+----
+
+NOTE: If you specify a `HTTP_CACERT_HASH`, https downloads are enabled and port 443 is used instead of port 80.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:100]:
```suggestion
When you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443 instead of HTTP over port 80.
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 10]
```diff
@@ -14,11 +14,15 @@ HTTP_HOST=downloads.raspberrypi.org
...
+`boot.img` and the `boot.sig` signature file is a ram disk containing a boot file system. For more details, see xref:raspberry-pi.adoc#boot_ramdisk[boot_ramdisk].
+
 HTTP in the `BOOT_ORDER` will be ignored if secure boot is enabled and xref:raspberry-pi.adoc#HTTP_HOST[HTTP_HOST] is not set.
 
 === Requirements
 
-To use HTTP boot, xref:raspberry-pi.adoc#bootloader_update_stable[update the bootloader] to a bootloader dated 10th March 2022 or later. HTTP boot only works over Ethernet, so you need to connect your Raspberry Pi to your network via an Ethernet cable, e.g. to a socket on the back of your router.
+To use HTTP boot, xref:raspberry-pi.adoc#bootloader_update_stable[update] to a bootloader released 10th March 2022 or later. HTTP boot requires a wired Ethernet connection.
+
+To use custom CA certificates, xref:raspberry-pi.adoc#bootloader_update_stable[update the bootloader] to a bootloader released 5th April 2024 or later. Only devices running the BCM2712 CPU support custom CA certificates.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:25]:
```suggestion
To use custom CA certificates, xref:raspberry-pi.adoc#bootloader_update_stable[update] to a bootloader released 5th April 2024 or later. Only devices running the BCM2712 CPU support custom CA certificates.
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 11]
```diff
@@ -40,6 +44,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+
+[gpio8=0]
+BOOT_ORDER=0xf7
+NET_INSTALL_ENABLED=0
+HTTP_HOST=<your_website>
+HTTP_PATH=<path_to_files>
+HTTP_CACERT_HASH=<hash>
+----
+
+When you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443 instead of HTTP over port 80.
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:102]:
```suggestion
When you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443. Without a hash, Network install downloads the image using HTTP over port 80.
```

File: documentation/asciidoc/computers/raspberry-pi/boot-http.adoc [hunk 12]
```diff
@@ -40,6 +44,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
...
+
+Finally, use the following commands to load everything into EEPROM:
+
+[source,console]
+----
+$ rpi-eeprom-config -c boot.conf -p mypubkey.pem -o pieeprom.bin --cacertder cert.der pieeprom.original.bin
+$ rpi-eeprom-digest -k myprivkey.pem -i pieeprom.bin -o pieeprom.sig
+----
+
+During network boot, your Raspberry Pi should use HTTPS instead of HTTP. You can check the boot output for the HTTPS URL:
```
Comment (nathan-contino) [inline documentation/asciidoc/computers/raspberry-pi/boot-http.adoc:112]:
```suggestion
During network boot, your Raspberry Pi should use HTTPS instead of HTTP. To see the full HTTPS URL resolved by Network Install for the download, check the boot output:
```

**Comment by peterharperuk** [comment]
Thanks @nathan-contino
